addon/components/async-await.js

```diff
@@ -145,14 +152,16 @@ export default Component.extend({
 
   didReceiveAttrs() {
     this._super(...arguments);
-    this.didReceivePromise(this.promise);
+    this.didReceiveArgument(this.argument);
   },
 
-  didReceivePromise(promise) {
-    if (promise === this.awaited) { return; }
+  didReceiveArgument(argument) {
+    if (argument === this.awaited) { return; }
+
+    this.promise = isHash(argument) ? RSVP.hash(argument) : Promise.resolve(argument);
```

Comment:
We don't need the `Promise.resolve` on the RHS, since we are about to do it on L173. (We do want to `Promise.resolve` the `RSVP.hash` to turn it into a real promise. I think we can also avoid introducing a new instance variable (`this.promise`) by passing `argument` to the callbacks instead. It is only used to check if "if I am still current", which can be done with `argument === this.awaited` now.